test/integration-tests/ui/PackageDependencyProvider.test.ts

```diff
@@ -13,107 +13,184 @@
-        expect(folder).to.not.be.undefined;
-
-        expect(folder?.path).to.equal(`${testAssetPath("defaultPackage")}/Sources`);
-
-        const childFolders = await treeProvider.getChildren(folder);
-        const childFolder = childFolders.find(n => n.name === "PackageExe");
-        expect(childFolder).to.not.be.undefined;
-
-        expect(childFolder?.path).to.equal(`${testAssetPath("defaultPackage")}/Sources/PackageExe`);
-
-        const files = await treeProvider.getChildren(childFolder);
-        const file = files.find(n => n.name === "main.swift");
-        expect(file).to.not.be.undefined;
-
-        expect(file?.path).to.equal(
-            `${testAssetPath("defaultPackage")}/Sources/PackageExe/main.swift`
-        );
-    });
-
-    test("Lists remote dependency file structure", async () => {
-        const items = await treeProvider.getChildren();
-
-        const dep = items.find(n => n.name === "swift-markdown") as PackageNode;
-        expect(dep).to.not.be.undefined;
-
-        const folders = await treeProvider.getChildren(dep);
-        const folder = folders.find(n => n.name === "Sources");
-        expect(folder).to.not.be.undefined;
-
-        const path = `${testAssetPath("dependencies")}/.build/checkouts/swift-markdown`;
-        expect(folder?.path).to.equal(`${path}/Sources`);
-
-        const childFolders = await treeProvider.getChildren(folder);
-        const childFolder = childFolders.find(n => n.name === "CAtomic");
-        expect(childFolder).to.not.be.undefined;
-
-        expect(childFolder?.path).to.equal(`${path}/Sources/CAtomic`);
-
-        const files = await treeProvider.getChildren(childFolder);
-        const file = files.find(n => n.name === "CAtomic.c");
-        expect(file).to.not.be.undefined;
-
-        expect(file?.path).to.equal(`${path}/Sources/CAtomic/CAtomic.c`);
+    suite("Full work flow tests", function () {
+        let folderContext: FolderContext;
+        let workspaceContext: WorkspaceContext;
+        const windowMock = mockGlobalObject(vscode, "window");
+
+        suiteSetup(async function () {
+            workspaceContext = await globalWorkspaceContextPromise;
+            await waitForNoRunningTasks();
+            folderContext = await folderContextPromise("dependencies");
+            await workspaceContext.focusFolder(folderContext);
+        });
+
+        test("Use local dependency", async () => {
+            // Expect to fail without setting up local version
+            const tasks = (await getBuildAllTask(folderContext)) as SwiftTask;
+            let { exitCode, output } = await executeTaskAndWaitForResult(tasks);
+            expect(exitCode).to.not.equal(0);
+            expect(output).to.include("PackageLib");
+            expect(output).to.include("required");
+
+            // Contract: spm reset, resolve, update should work
+            const executeTaskSpy = sinon.spy(utilities, "executeTaskWithUI");
```

Comment:
this test gets into needing some intimate knowledge of the implementation for spying / mocking. If this is tied to a command, we should execute the command. Instead of mocking the user selection, you can let the command take a parameter and pass that parameter when running executeCommand

Reply:
Done, switched to assert on executeCommand.